apps/tx-builder/src/components/forms/fields/JsonField.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState, useCallback, ClipboardEvent } from 'react';
+import styled from 'styled-components';
+import { Icon, TextFieldInput, Tooltip, GenericModal, Text, Button, IconTypes } from '@gnosis.pm/safe-react-components';
+import IconButton from '@material-ui/core/IconButton';
+import { Box } from '@material-ui/core';
+
+const DEFAULT_ROWS = 4;
+
+type Props = {
+  id: string;
+  name: string;
+  label: string;
+  value: string;
+  onChange: (value: string) => void;
+};
+
+const JsonField = ({ id, name, label, value, onChange }: Props) => {
+  const [showReplaceModal, setShowReplaceModal] = useState(false);
+  const [tempAbi, setTempAbi] = useState(value);
+  const [isPrettified, setIsPrettified] = useState(false);
+
+  const toggleFormatJSON = useCallback(() => {
+    if (!value) {
+      return;
+    }
+
+    try {
+      onChange?.(JSON.stringify(JSON.parse(value), null, isPrettified ? 0 : 2));
```

Comment:
The type for onChange doesn't allow undefined, why do we do `?` here? It's the compiler's job to catch if we pass `undefined` here

Reply:
yeah you are right. I think this remained from a previous implementation i tried